smurfs/src/actions/Action.js

```diff
@@ -0,0 +1,48 @@
+export const FETCH_SMURF_START = "FETCH_SMURF_START";
+export const FETCH_SMURF_SUCCESS = "FETCH_SMURF_SUCCESS";
+export const FETCH_SMURF_FAIL = "FETCH_SMURF_FAIL";
+
+
+export const getSmurf = () => dispatch => {
+    
+    // console.log('start getSmurf');
+    dispatch({ type: FETCH_SMURF_START });
+    fetch('http://localhost:3333/smurfs', {
+        "method": "GET",
+    })
+        .then(response => {
+            return response.json();
+        })
+        .then(res => {
+            // console.log(res);
+            dispatch({ type: FETCH_SMURF_SUCCESS, payload: res })
+        })
+        .catch (err => {
+            dispatch({ type: FETCH_SMURF_FAIL, payload: err.response})
+        });
+};
+
+export const addSmurf = (item) => dispatch => {
+    
+    console.log('start addSmurf');
+    // dispatch({ type: FETCH_SMURF_START });
+    fetch('http://localhost:3333/smurfs', {
+        "method": "POST",
+        body: JSON.stringify(item),
+        headers:{
+            'Content-Type': 'application/json'
+          }
+        }).then(res => res.json())
+        // .then(response => console.log('Success:', JSON.stringify(response)))
+        
+
+        // .then(response => {
+        //     return response.json();
+        // })
+        .then(res => {
+            console.log(res);
+            dispatch({ type: FETCH_SMURF_SUCCESS, payload: res })
+        })
+
+        .catch(error => console.error('Error:', error));
+};
```

Comment:
Lots of stuff here. For this sort of action creator you'll typically see `ADD_SMURF_START/SUCCESS/FAILURE` being called in their respective places. Also, I see you are calling `FETCH_SMURF_SUCCESS` inside `addSmurf` when it was built for `getSmurf`. While this works perfectly fine it seems extra sloppy since you are pimping out `getSmurf` with it's own custom actions and then gypping `addSmurf` by reusing others inside of it. The real issue here is consistency in your code which is a very attractive characteristic for your code to have.